src/AutoRest.CSharp/Common/AutoRest/Communication/StandaloneGeneratorRunner.cs

```diff
@@ -12,46 +12,71 @@
 using AutoRest.CSharp.AutoRest.Plugins;
 using AutoRest.CSharp.Common.Input;
 using AutoRest.CSharp.Input;
+using AutoRest.CSharp.Output.Models.Types;
 using Azure.Core;
+using Microsoft.CodeAnalysis;
 
 namespace AutoRest.CSharp.AutoRest.Communication
 {
     internal class StandaloneGeneratorRunner
     {
-        public static async Task RunAsync(string[] args)
+        public static async Task RunAsync(CommandLineOptions options)
         {
-            var basePath = args.Single(a => !a.StartsWith("--"));
+            string? projectPath = null;
+            string outputPath;
+            bool wasProjectPathPassedIn = options.ProjectPath is not null;
+            if (options.Standalone is not null)
+            {
+                //TODO this is only here for back compat we should consider removing it
+                outputPath = options.Standalone;
+            }
+            else
+            {
+                projectPath = options.ProjectPath!;
+                outputPath = Path.Combine(projectPath, "Generated");
+            }
 
-            LoadConfiguration(basePath, File.ReadAllText(Path.Combine(basePath, "Configuration.json")));
+            var configurationPath = options.ConfigurationPath ?? Path.Combine(outputPath, "Configuration.json");
```

Comment:
Configuration.json is an template file which will be generated by the autorest/cadl-emitter according to the autorest configure/cadl-options-or-sidecar. We will not allow customer to provide a Configuration.json file.

Reply:
Yes, this is a temporary solution before sidecar is ready. Currently we need to expose this configuration thing to users to let them do some setting.